# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/wasm/spec/func.wast.js
# Commit: e71e54d93af8
# Full Hash: e71e54d93af8baa6ef915c81dc04ac3400f825cd
# Author: Andy Wingo <wingo@igalia.com>
# Date: 2020-03-06 20:10:41
# Regressor Bug: 1620197
# File Overlap Count: 1
# Description:
#   Bug 1620197 - Enable multiple results from WebAssembly functions r=lth
#   
#   This patch enables multi-value calls and returns, adding some tests, and
#   conditionally disabling a couple expect-fail reftests that now pass.
#   
# ==============================================================================

diff -r afbf73fb4073 -r e71e54d93af8 js/src/jit-test/tests/wasm/spec/func.wast.js
--- a/js/src/jit-test/tests/wasm/spec/func.wast.js	Fri Mar 06 09:57:45 2020 +0000
+++ b/js/src/jit-test/tests/wasm/spec/func.wast.js	Fri Mar 06 13:53:07 2020 +0000
@@ -278,11 +278,17 @@
 // func.wast:484
 assert_invalid("\x00\x61\x73\x6d\x01\x00\x00\x00\x01\x86\x80\x80\x80\x00\x01\x60\x02\x7c\x7e\x00\x03\x82\x80\x80\x80\x00\x01\x00\x0a\x8b\x80\x80\x80\x00\x01\x85\x80\x80\x80\x00\x00\x20\x01\x9a\x0b");
 
+// These two tests check that function types returning more than 2
+// results are invalid, but with multi-value of course that's no longer
+// the case.  Until the feature fully lands and we can import from
+// upstream, skip these tests.
+if (!wasmMultiValueEnabled()) {
 // func.wast:492
 assert_invalid("\x00\x61\x73\x6d\x01\x00\x00\x00\x01\x86\x80\x80\x80\x00\x01\x60\x00\x02\x7f\x7f\x03\x82\x80\x80\x80\x00\x01\x00\x0a\x89\x80\x80\x80\x00\x01\x83\x80\x80\x80\x00\x00\x00\x0b");
 
 // func.wast:496
 assert_invalid("\x00\x61\x73\x6d\x01\x00\x00\x00\x01\x86\x80\x80\x80\x00\x01\x60\x00\x02\x7f\x7f\x03\x82\x80\x80\x80\x00\x01\x00\x0a\x89\x80\x80\x80\x00\x01\x83\x80\x80\x80\x00\x00\x00\x0b");
+}
 
 // func.wast:505
 assert_invalid("\x00\x61\x73\x6d\x01\x00\x00\x00\x01\x85\x80\x80\x80\x00\x01\x60\x00\x01\x7f\x03\x82\x80\x80\x80\x00\x01\x00\x0a\x88\x80\x80\x80\x00\x01\x82\x80\x80\x80\x00\x00\x0b");